Library/Homebrew/formula_installer.rb

```diff
@@ -1400,11 +1413,12 @@ def forbidden_tap_check
       compute_dependencies.each do |(dep, _options)|
         dep_tap = dep.tap
         next if dep_tap.blank?
-        next unless forbidden_taps_set.include?(dep_tap)
+        next if allowed_tap?(dep_tap, allowed_taps_set, forbidden_taps_set)
 
         raise CannotInstallFormulaError, <<~EOS
           The installation of #{formula.name} has a dependency #{dep.name}
-          but the #{dep_tap} tap was forbidden by #{owner} in `HOMEBREW_FORBIDDEN_TAPS`.#{owner_contact}
+          but #{owner} has either not allowed the #{dep_tap} tap in `HOMEBREW_ALLOWED_TAPS` or
+          has forbidden the #{dep_tap} tap in `HOMEBREW_FORBIDDEN_TAPS`.#{owner_contact}
```

Comment:
I wonder if having this be more specific (telling them whether is it not allowed or have been forbidden) would be helpful?

Reply:
I do think it would be more helpful -- I just got stuck on a nice way of doing it 😄